[WS Regression] eth_newFilter is failing

### Description

The test cases below are failing.
```bash
RPC Server Acceptance Tests
       Acceptance tests
         @web-socket-batch-2 eth_newFilter
           Standard Web Socket
             @release Should execute eth_newFilter on Standard Web Socket and handle valid requests correctly:
     AssertionError: expected undefined to exist
      at Suite.<anonymous> (packages/ws-server/tests/acceptance/newFilter.spec.ts:90:26)
      at Generator.next (<anonymous>)
      at fulfilled (packages/ws-server/tests/acceptance/newFilter.spec.ts:24:58)
```


```bash
RPC Server Acceptance Tests
       Acceptance tests
         @web-socket-batch-2 eth_newFilter
           Ethers Web Socket Provider
             @release Should execute eth_newFilter on Ethers Web Socket Provider and handle valid requests correctly:
     Error: could not coalesce error (error={ "code": -32000, "message": "Exceeded maximum block range: 1000" }, payload={ "id": 1, "jsonrpc": "2.0", "method": "eth_newFilter", "params": [ { "address": [ "0xadF86D6D774AA2507380E0264e5f8052035B5BE8" ], "fromBlock": "0x0", "toBlock": "latest" } ] }, code=UNKNOWN_ERROR, version=6.13.2)
      at makeError (packages/ws-server/node_modules/ethers/src.ts/utils/errors.ts:694:21)
      at WebSocketProvider.getRpcError (packages/ws-server/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:1059:25)
      at /Users/user/development/hashgraph/hedera-json-rpc-relay/packages/ws-server/node_modules/ethers/src.ts/providers/provider-jsonrpc.ts:563:45
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Steps to reproduce

1. Run `npm run release_ws`
2. Observe the failing test

### Additional context

_No response_

### Hedera network

testnet, previewnet

### Version

0.58.0-rc1

### Operating system

None

--- a/packages/ws-server/src/controllers/index.ts
+++ b/packages/ws-server/src/controllers/index.ts
@@ -81,6 +81,7 @@ const handleSendingRequestsToRelay = async ({
       chainId: (_, requestDetails) => [requestDetails],
       estimateGas: (params, requestDetails) => [...params, null, requestDetails],
       getStorageAt: (params, requestDetails) => [params[0], params[1], requestDetails, params[2]],
+      newFilter: (params, requestDetails) => [params[0], params[1], requestDetails, params[2], params[3]],
       default: (params, requestDetails) => [...params, requestDetails],
     };
 

--- a/packages/ws-server/src/controllers/index.ts
+++ b/packages/ws-server/src/controllers/index.ts
@@ -26,7 +26,7 @@ import { handleEthSubscribe, handleEthUnsubscribe } from './eth_subscribe';
 import { JsonRpcError, predefined, Relay } from '@hashgraph/json-rpc-relay';
 import { MirrorNodeClient } from '@hashgraph/json-rpc-relay/dist/lib/clients';
 import jsonResp from '@hashgraph/json-rpc-server/dist/koaJsonRpc/lib/RpcResponse';
-import { resolveParams, validateJsonRpcRequest, verifySupportedMethod } from '../utils/utils';
+import { paramRearrangementMap, resolveParams, validateJsonRpcRequest, verifySupportedMethod } from '../utils/utils';
 import {
   InvalidRequest,
   IPRateLimitExceeded,
@@ -76,17 +76,8 @@ const handleSendingRequestsToRelay = async ({
     const resolvedParams = resolveParams(method, params);
     const [service, methodName] = method.split('_');
 
-    // Rearrange the parameters for certain methods, since not everywhere requestDetails is last aparameter
-    const paramRearrangementMap: { [key: string]: (params: any[], requestDetails: RequestDetails) => any[] } = {
-      chainId: (_, requestDetails) => [requestDetails],
-      estimateGas: (params, requestDetails) => [...params, null, requestDetails],
-      getStorageAt: (params, requestDetails) => [params[0], params[1], requestDetails, params[2]],
-      newFilter: (params, requestDetails) => [params[0], params[1], requestDetails, params[2], params[3]],
-      default: (params, requestDetails) => [...params, requestDetails],
-    };
-
-    const rearrangeParams = paramRearrangementMap[methodName] || paramRearrangementMap['default'];
-    const rearrangedParams = rearrangeParams(resolvedParams, requestDetails);
+    const rearrangeParamsFn = paramRearrangementMap[methodName] || paramRearrangementMap['default'];
+    const rearrangedParamsArray = rearrangeParamsFn(resolvedParams, requestDetails);
 
     // Call the relay method with the resolved parameters.
     // Method will be validated by "verifySupportedMethod" before reaching this point.
@@ -95,9 +86,9 @@ const handleSendingRequestsToRelay = async ({
       txRes = await relay
         .eth()
         .filterService()
-        [methodName](...rearrangedParams);
+        [methodName](...rearrangedParamsArray);
     } else {
-      txRes = await relay[service]()[methodName](...rearrangedParams);
+      txRes = await relay[service]()[methodName](...rearrangedParamsArray);
     }
 
     if (!txRes) {

--- a/packages/ws-server/src/utils/utils.ts
+++ b/packages/ws-server/src/utils/utils.ts
@@ -202,3 +202,18 @@ export const constructValidLogSubscriptionFilter = (filters: any): object => {
     Object.entries(filters).filter(([key, value]) => value !== undefined && ['address', 'topics'].includes(key)),
   );
 };
+
+/**
+ * A mapping of parameter rearrangement functions for various methods.
+ * Each function adjusts the order of parameters to ensure that the
+ * `requestDetails` object is placed correctly based on the method's requirements.
+ */
+export const paramRearrangementMap: {
+  [key: string]: (params: any[], requestDetails: RequestDetails) => any[];
+} = {
+  chainId: (_: any[], requestDetails: RequestDetails) => [requestDetails],
+  estimateGas: (params, requestDetails) => [params[0], params[1], requestDetails],
+  getStorageAt: (params, requestDetails) => [params[0], params[1], requestDetails, params[2]],
+  newFilter: (params, requestDetails) => [params[0], params[1], requestDetails, params[2], params[3]],
+  default: (params, requestDetails) => [...params, requestDetails],
+};

--- a/packages/ws-server/tests/unit/utils.spec.ts
+++ b/packages/ws-server/tests/unit/utils.spec.ts
@@ -26,6 +26,7 @@ import {
   getMultipleAddressesEnabled,
   getWsBatchRequestsEnabled,
   handleConnectionClose,
+  paramRearrangementMap,
   resolveParams,
   sendToClient,
 } from '../../src/utils/utils';
@@ -310,6 +311,58 @@ describe('Utilities unit tests', async function () {
     WsTestHelper.withOverriddenEnvsInMochaTest({ WS_BATCH_REQUESTS_MAX_SIZE: undefined }, () => {
       it('should return 20', () => {
         expect(getBatchRequestsMaxSize()).to.equal(20);
+      });
+    });
+  });
+
+  describe('paramRearrangementMap', () => {
+    const requestDetails = new RequestDetails({ ipAddress: '0.0.0.0', requestId: 'test-id' });
+    const specialMethodNames = [`chainId`, `estimateGas`, `getStorageAt`, `newFilter`, `default`];
+
+    const mockResolvedParams = {
+      chainId: [],
+      estimateGas: [
+        {
+          to: '0xD7d454ea421FA3E98c988c2A33b5292C70A43b1E',
+          data: '0x18160ddd',
+        },
+        'latest',
+      ],
+      getStorageAt: ['0xd7d454ea421fa3e98c988c2a33b5292c70a43b1e', '0x0', 'latest'],
+      newFilter: [
+        '0x0',
+        'latest',
+        ['0xf72ea4E404618E9DCcA79748236910887be9e2bd'],
+        ['0x25d719d88a4512dd76c7442b910a83360845505894eb444ef299409e180f8fb9'],
+      ],
+      default: ['0x7cb9357e', '0x7cb9357e', '0x00abv'],
+    };
+
+    const expectedRearrangedParams = {
+      chainId: [requestDetails],
+      estimateGas: [...mockResolvedParams.estimateGas, requestDetails],
+      getStorageAt: [
+        mockResolvedParams.getStorageAt[0],
+        mockResolvedParams.getStorageAt[1],
+        requestDetails,
+        mockResolvedParams.getStorageAt[2],
+      ],
+      newFilter: [
+        mockResolvedParams.newFilter[0],
+        mockResolvedParams.newFilter[1],
+        requestDetails,
+        mockResolvedParams.newFilter[2],
+        mockResolvedParams.newFilter[3],
+      ],
+      default: [...mockResolvedParams.default, requestDetails],
+    };
+
+    specialMethodNames.forEach((methodName) => {
+      it(`Should correctly rearrange parameters for ${methodName}`, () => {
+        const rearrangeParamsFn = paramRearrangementMap[methodName];
+        const rearrangedParamsArray = rearrangeParamsFn(mockResolvedParams[methodName], requestDetails);
+        const expectedResult = expectedRearrangedParams[methodName];
+        expect(rearrangedParamsArray).to.deep.eq(expectedResult);
       });
     });
   });

